### PR TITLE
Add command to trim streams

### PR DESCRIPTION
**Pull request by wachterjohannes**

### CODE REVIEW

File: Command/TrimRedisStreamCommand.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        $auth = $parsedUrl['user'] ?? null;
+
+        $redis = new \Redis();
+        $redis->connect($parsedUrl['host'], $parsedUrl['port']);
+        if ($auth) {
+            $redis->auth($auth);
+        }
+
+        $x = $redis->xTrim($stream, (int) $maxLength, true);
+
```
Comment (alexander-schranz) [inline Command/TrimRedisStreamCommand.php:78]: add also:

```php
if ($errorMessage = $redis->getLastError()) {
    throw new \RuntimeException($errorMessage);
}
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: Command/TrimRedisStreamCommand.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        $stream = $dsnParts[1];
+        $auth = $parsedUrl['user'] ?? null;
+
+        $redis = new \Redis();
+        $redis->connect($parsedUrl['host'], $parsedUrl['port']);
+        if ($auth) {
+            $redis->auth($auth);
+        }
+
+        $x = $redis->xTrim($stream, (int) $maxLength, true);
```
Comment (alexander-schranz) [inline Command/TrimRedisStreamCommand.php]: xtrim need to be lowercase else prophecy will fail because its case sensitive

**Comment by alexander-schranz** [comment]
any way to provide a test for this command?

**Comment by wachterjohannes** [comment]
@alexander-schranz we would have to create a functional testcase - is there already a setup for this?

File: Command/TrimRedisStreamCommand.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class TrimRedisStreamCommand extends Command
+{
+    protected static $defaultName = 'redis-transport:trim';
+
+    /**
+     * @var \Redis
```
Comment (alexander-schranz) [inline Command/TrimRedisStreamCommand.php]:
```
@var \Redis|null
```

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**